Tests/LibWeb/Text/expected/URL/strip_trailing_spaces_from_opaque_path.txt

```diff
@@ -0,0 +1,4 @@
+pathname => '/well    '
+pathname => '/well'
```

Comment:
The results for this test don't match the results which I see in chrome or firefox. I believe that this is a bug in our implementation.

A space in the spec is in the percent encoding for "Path", and I believe that we are percent encoding here: https://github.com/SerenityOS/serenity/blob/ec416563a896f188072bac6e86a3234888a8f231/AK/URLParser.cpp#L1539

I think that the issue is that we are percent decoding the path in Web/URL.cpp, which is producing wrong results.

Reply:
Not sure if this is the problem, but I do see that we percent decode in `AK::serialize_path()`:

https://github.com/SerenityOS/serenity/blob/ec416563a896f188072bac6e86a3234888a8f231/AK/URL.cpp#L270

Reply:
Yup, I think that is the problem. Inside of web specs we should probably not be doing the percent decode. Outside of it, we probably should

Reply:
I'm pretty sure this is the relevant [spec ](https://url.spec.whatwg.org/#url-path-serializer)to `AK::URL::serialize_path()`. There's no explicit mention of a need to percent decode the path during serialization

Reply:
Continuing my previous [comment](https://github.com/SerenityOS/serenity/pull/21021#issuecomment-1713346790), I also tried to just outright ignore the first step of this method, but that too causes the test suite to hang for some reason.

Oddly enough, reverting the first step back to its old self (i.e. using `cannot_be_a_base_url()`) does not cause any test suite hangs.

Reply:
One other note; I forgot to account for the URL tests in `AK`. We seem to fail in quite a few: mainly because of the missing slash prefix issue in step 1 and characters being escaped (I tested the failing `unicode` test case specifically on Chrome and Firefox and they escape their pathnames, which is not what we expect in this test case).

If you wish, we could try to tackle these issues within another PR or I could use my workaround of reverting step one of `serialize_path()` back to using `cannot_be_a_base_url()` since this doesn't cause the `LibWeb` test suite to hang.